File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-1.cpp
class Solution {
public:
    void solve(int index, string temp, string digits, unordered_map<char,string> mp, vector<string> &res){
        if(temp.length() == digits.length()){
            res.push_back(temp);
            return;
        }
        for(char c: mp[digits[index]]){
            temp.push_back(c);
            solve(index+1, temp, digits, mp,res);
            temp.pop_back();
        }
    }
    vector<string> letterCombinations(string digits) {
        vector<string> res;
        if(digits.length() == 0){
            return res;
        }
        unordered_map<char, string> mp = {
            {'2' , "abc"},
            {'3' , "def"},
            {'4' , "ghi"},
            {'5' , "jkl"},
            {'6' , "mno"},
            {'7' , "pqrs"},
            {'8' , "tuv"},
            {'9' , "wxyz"},
        };
        solve(0, "", digits, mp, res);
        return res;

    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        res = ""
        for i in range(len(s)):
            l = i-1
            h = i+1
            temp = ""
            temp += s[i]
            while l >= 0 and h < len(s) and s[l] == s[h]:
                temp = s[l] + temp + s[h]
                l -= 1
                h += 1
            if len(temp) > len(res):
                res = temp
            l = i
            h = i + 1
            temp = ""
            while l >= 0 and h < len(s) and s[l] == s[h]:
                temp = s[l] + temp + s[h]
                l -= 1
                h += 1
            if len(temp) > len(res):
                res = temp
        return res